frontends/p4/sideEffects.cpp

```diff
@@ -169,6 +203,7 @@ class DismantleExpression : public Transform {
         visit(expression->expr);
         auto left = result->final;
         CHECK_NULL(left);
+        // TODO: why clone here?
```

Comment:
If you don't clone you have all sorts of weird problems (which are also a big pain to debug, BTW). For example, consider def-use analysis: the same expression has different uses and definitions depending of where it occurs in the program. So although something is syntactically identical, it should be represented by a different IR node. You will want different attributes computed for the two clones.